lib/internal/url.js

```diff
@@ -1504,50 +1502,39 @@ function fileURLToPath(path) {
 // - CR: The carriage return character is also stripped out by the `pathname`
 //       setter.
 // - TAB: The tab character is also stripped out by the `pathname` setter.
-const percentRegEx = /%/g;
-const backslashRegEx = /\\/g;
-const newlineRegEx = /\n/g;
-const carriageReturnRegEx = /\r/g;
-const tabRegEx = /\t/g;
-
 function encodePathChars(filepath) {
-  if (StringPrototypeIncludes(filepath, '%'))
-    filepath = StringPrototypeReplace(filepath, percentRegEx, '%25');
+  filepath = StringPrototypeReplaceAll(filepath, '%', '%25');
   // In posix, backslash is a valid character in paths:
-  if (!isWindows && StringPrototypeIncludes(filepath, '\\'))
-    filepath = StringPrototypeReplace(filepath, backslashRegEx, '%5C');
-  if (StringPrototypeIncludes(filepath, '\n'))
-    filepath = StringPrototypeReplace(filepath, newlineRegEx, '%0A');
-  if (StringPrototypeIncludes(filepath, '\r'))
-    filepath = StringPrototypeReplace(filepath, carriageReturnRegEx, '%0D');
-  if (StringPrototypeIncludes(filepath, '\t'))
-    filepath = StringPrototypeReplace(filepath, tabRegEx, '%09');
+  if (!isWindows) filepath = StringPrototypeReplaceAll(filepath, '\\', '%5C');
+  filepath = StringPrototypeReplaceAll(filepath, '\n', '%0A');
+  filepath = StringPrototypeReplaceAll(filepath, '\r', '%0D');
+  filepath = StringPrototypeReplaceAll(filepath, '\t', '%09');
   return filepath;
 }
 
 function pathToFileURL(filepath) {
   const outURL = new URL('file://');
   if (isWindows && StringPrototypeStartsWith(filepath, '\\\\')) {
     // UNC path format: \\server\share\resource
-    const paths = StringPrototypeSplit(filepath, '\\');
-    if (paths.length <= 3) {
+    const hostnameEndIndex = StringPrototypeIndexOf('\\', 2);
```

Comment:
I didn't understand this line. Can you explain?

Reply:
We are looking for the (index of the) first `\` char after index 2. E.g. in `\\server\share\resource`, the hostname is `server`.